src/miniscript/satisfy.rs

```diff
@@ -34,28 +34,107 @@ use Miniscript;
 use ScriptContext;
 use Terminal;
 
-/// Type alias for a signature/hashtype pair
-pub type BitcoinSig = (secp256k1::Signature, bitcoin::SigHashType);
+/// type alias for legacy sighash type
+pub type LegacySigHashType = bitcoin::SigHashType;
+/// type alias for schnorr signature hash type (includes sighash default)
+pub type SchnorrSigHashType = bitcoin::util::sighash::SigHashType;
+/// Type for a EC signature/hashtype pair
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct BitcoinECSig {
```

Comment:
Btw we will have this type in `rust-bitcoin`: https://github.com/rust-bitcoin/rust-bitcoin/pull/669

As for the name, probably it should have `ECDSA` part (not simply `EC`)

Reply:
Good catch